qlty-analysis/src/workspace_entries/matchers.rs

```diff
@@ -234,6 +234,43 @@ fn globs_to_globset(globs: &[String]) -> Result<GlobSet> {
     Ok(builder.build()?)
 }
 
+#[derive(Default, Debug)]
+pub struct IgnoreGroupsMatcher {
+    matchers: Vec<GlobsMatcher>,
+}
+
+impl IgnoreGroupsMatcher {
+    pub fn new(ignore_groups: Vec<IgnoreGroup>) -> Self {
+        let matchers = ignore_groups
+            .into_iter()
+            .filter_map(|ignore_group| {
+                GlobsMatcher::new_for_globs(&ignore_group.ignores, ignore_group.negate).ok()
+            })
+            .collect();
+
+        Self { matchers }
+    }
+}
+
+impl WorkspaceEntryMatcher for IgnoreGroupsMatcher {
+    fn matches(&self, workspace_entry: WorkspaceEntry) -> Option<WorkspaceEntry> {
+        // By default, include the file
+        let path = workspace_entry.path_string();
+
+        for matcher in self.matchers.iter().rev() {
+            if matcher.glob_set.is_match(&path) {
+                return if matcher.include {
+                    Some(workspace_entry) // No need to clone; directly return ownership
+                } else {
+                    None // Immediately exclude the file if a normal ignore rule matches
+                };
+            }
+        }
+
+        Some(workspace_entry)
+    }
+}
+
```

Comment:
This matchers.rs file is getting pretty big. I recommend we make a `matchers/` folder and break it up...

This new IgnoreGroupsMatcher likely deserves it's own Rust file with inline tests, especially since the iter-reverse-match-none-some algorithm is quite tricky.

For the existing matches, I could imagine one file for the boolean (And + Any) matches, one for the Language-related matches (Language Globs, Language Shebang), not sure about the best way to organize the rest.